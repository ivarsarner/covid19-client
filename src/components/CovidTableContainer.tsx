import React, { useEffect, useState } from 'react';
import axios from 'axios';
import CovidTable from './CovidTable';
import CssBaseline from '@material-ui/core/CssBaseline';
import Container from '@material-ui/core/Container';
import Header from './Header';
import Loader from './Loader';

const initialState = {
  status: 'loading',
  error: '',
  americanStates: [],
};

const CovidTableContainer: React.FC = () => {
  const [covidData, setCovidData] = useState(initialState);
  const { status, error, americanStates } = covidData;

  const getCovidData = async () => {
    try {
      const url = process.env.REACT_APP_API_URL || '';
      const { data } = await axios.get(url);
      setCovidData({
        status: 'success',
        error: '',
        americanStates: data,
      });
    } catch (error) {
      setCovidData({
        status: 'error',
        error: error.message,
        americanStates: [],
      });
    }
  };

  useEffect(() => {
    getCovidData();
  }, []);

  return (
    <>
      <CssBaseline />
      <main>
        <Container maxWidth="md">
          <Header />
          {status === 'loading' ? (
            <>
              <Loader />
            </>
          ) : status === 'error' ? (
            <div>
              An error has occured: <br></br>
              {error}
            </div>
          ) : status === 'success' ? (
            <CovidTable americanStates={americanStates} />
          ) : (
            <div>No data found</div>
          )}
        </Container>
      </main>
    </>
  );
};

export default CovidTableContainer;
