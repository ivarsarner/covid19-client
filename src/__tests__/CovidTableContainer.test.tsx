import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CovidTableContainer from '../components/CovidTableContainer';

let container: any;
let getByText: any;

beforeEach(() => {
  const wrapper = render(<CovidTableContainer />);
  container = wrapper.container;
  getByText = wrapper.getByText;
});

afterEach(cleanup);

test('renders CovidTableContainer', () => {
  expect(container).toMatchSnapshot();
});

test('renders loading', () => {
  const loadingDiv = getByText('Loading...');
  expect(loadingDiv).toBeInTheDocument();
});
