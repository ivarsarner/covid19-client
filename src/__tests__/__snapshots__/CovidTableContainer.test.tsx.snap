// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CovidTableContainer 1`] = `
<div>
  <main>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div
        class="MuiContainer-root makeStyles-headerContent-9 MuiContainer-maxWidthLg"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
        >
          Covid-19 cases in USA
        </h4>
        <p
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
        >
          The table below uses data from 
          <a
            href="https://covidtracking.com"
          >
            The COVID Tracking Project
          </a>
           to show the amount of hospitalized patients in USA, as well as new deaths in the last 3 days.
        </p>
      </div>
      <div
        class="makeStyles-root-40"
      >
        <div>
          Loading...
        </div>
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </main>
</div>
`;
